### PR TITLE
Update user statistics models in line with web API changes

### PR DESCRIPTION
**Pull request by frenzibyte**
See https://github.com/ppy/osu-web/pull/7245.

1. Moves away from using `rank.country` as it's deprecated and will be removed soon, and also nukes the whole `UserRanks` object in favour of those outer JSON properties.
2. Nukes no longer existing `current_mode_rank` and use the now included `statistics` property for friends display.

Also makes tournament client refetch users on null country rank, may not be a nice thing, but seems to be the easiest way to do the migration for brackets file as done with global rank API change, not sure if it deserves an inline comment?

### CODE REVIEW

**Comment by bdach** [comment]
> Also makes tournament client refetch users on null country rank, may not be a nice thing, but seems to be the easiest way to do the migration for brackets file as done with global rank API change, not sure if it deserves an inline comment?

I don't get this. What are you saying exactly?

**Comment by frenzibyte** [comment]
Just like done previously with global rank. Changing serializable members of `UserStatistics` will break `bracket.json` from deserializing correctly, therefore I've made the tournament client refetch user for simplicity when `CountryRank` is not deserialized correctly (by checking if null).

**Comment by bdach** [comment]
Wasn't the `UserRanks` thing doing the same? Anyway, if it's temporary code to be removed at a later date, it should be marked as such.

**Comment by frenzibyte** [comment]
No? `ranks.global` was straight up removed and replaced with `global_rank`, this PR is doing so with `ranks.country`.

The point of that private `UserRanks` setter IIRC was nothing more than just better code readability (i.e. not having `GlobalRank` and `Ranks.Country` side to side), and also for getting ready of `UserRanks` removal.

File: osu.Game.Tournament/TournamentGameBase.cs [hunk 1]
```diff
@@ -150,7 +150,9 @@ private bool addPlayers()
             {
                 foreach (var p in t.Players)
                 {
-                    if (string.IsNullOrEmpty(p.Username) || p.Statistics?.GlobalRank == null)
+                    if (string.IsNullOrEmpty(p.Username)
+                        || p.Statistics?.GlobalRank == null
```
Comment (peppy) [inline osu.Game.Tournament/TournamentGameBase.cs:154]: aren't these nullable? couldn't this trigger a repopulation on every startup?

Comment (frenzibyte) [reply osu.Game.Tournament/TournamentGameBase.cs:154]: Yeah that's one down of this solution, which happened to `GlobalRank` as well. I would say it's unlikely to add users who don't have global ranks in tournament teams logically. 

But I believe that's not a valid reason, one alternative solution would be a custom JSON converter for deserializing user that would aim at reading from `rank` object and migrate to the new outer properties.

**Comment by frenzibyte** [comment]
For what it's worth, sorting by ranks in friend overlay is broken in `master`, due to the removal of `current_mode_rank` web-side, which this PR migrates away from by using `Statistics.GlobalRank` instead. Can separate the migration of current mode rank to another PR if the other changes still need some thought.

File: osu.Game.Tournament/TournamentGameBase.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Tournament/TournamentGameBase.cs:154]: Let's just go with this for now. I agree it will probably not be a common occurrence, if ever.

**Review by peppy — APPROVED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**